File: write_everyday/917.仅仅反转字母.py
class Solution:
    def reverseOnlyLetters(self, s: str) -> str:
        char , begin, end = list(s), 0, len(s)-1
        while(begin < end):
            while begin < end and not ('a' <= s[begin] <= 'z' or 'A' <= s[begin] <= 'Z'):
                begin+=1
            while begin < end and not ('a' <= s[end] <= 'z' or 'A' <= s[end] <= 'Z'):
                end-=1
            if begin < end:
                char[begin], char[end] = char[end], char[begin]
                begin+=1
                end-=1
        return ''.join(char)

File: write_everyday/917.仅仅反转字母.go
func reverseOnlyLetters(s string) string {
    bytes := []byte(s)
     
    for l,r := 0, len(s)-1 ; l<r ; {
        for l < r && !((bytes[l] >= 'a' && bytes[l] <= 'z') || (bytes[l] >= 'A' && bytes[l] <= 'Z')) {
            l++
        }
        for l<r && !(( bytes[r] >= 'a' && bytes[r] <= 'z' ) || (bytes[r]  >= 'A' && bytes[r]  <= 'Z' )) {
            r--
        }
        if l<r {
            bytes[l], bytes[r] = bytes[r], bytes[l]
            l++
            r--
        }
    }
    return string(bytes)
}